Model/Queue/Processing.php

```diff
@@ -373,119 +393,97 @@ protected function processWithAvaTax(Queue $queue, $entity)
     /**
      * @param \Magento\Sales\Api\Data\InvoiceInterface|\Magento\Sales\Api\Data\CreditmemoInterface $entity
      * @param \ClassyLlama\AvaTax\Api\Data\GetTaxResponseInterface $processSalesResponse
+     * @throws \Exception
      */
-    protected function updateAdditionalEntityAttributes(
+    protected function saveAvaTaxRecord(
         $entity,
         GetTaxResponseInterface $processSalesResponse
-    ) {
-        $entityExtension = $entity->getExtensionAttributes();
-        if ($entityExtension == null) {
-            $entityExtension = $this->getEntityExtensionInterface($entity);
-        }
+    )
+    {
+        $avaTaxRecord = $this->createAvataxEntity($entity);
+        // Load existing AvaTax entry for this entity, if exists
+        $avaTaxRecord->loadByParentId($entity->getId());
```

Comment:
Another place where the `loadByParentId()` method could be eliminated in favor of load($entity->getId(), 'parent_id')`